File: 36 - Longest Increasing Subsequence in an Array/solution.py
def length_of_lis(nums):
    # Write your code here
    if not nums:
        return 0

    dp = [1] * len(nums)

    for i in range(1, len(nums)):
        for j in range(0, i):
            if nums[j] < nums[i]:
                dp[i] = max(dp[i], dp[j] + 1)

    return max(dp)


# Test Cases
print(length_of_lis([10, 9, 2, 5, 3, 7, 101, 18]))

File: 36 - Longest Increasing Subsequence in an Array/solution.js
function lengthOfLIS(nums) {
  // Write your code here
  if (!nums.length) return 0;

  const dp = new Array(nums.length).fill(1);

  for (let i = 1; i < nums.length; i++) {
    for (let j = 0; j < i; j++) {
      if (nums[j] < nums[i]) {
        dp[i] = Math.max(dp[i], dp[j] + 1);
      }
    }
  }

  return Math.max(...dp);
}

// Test Cases
console.log(lengthOfLIS([10, 9, 2, 5, 3, 7, 101, 18]));
